Spec/Test Utilities/TestUtilities.swift

```diff
@@ -46,6 +46,26 @@ let testTimeout = DispatchTimeInterval.seconds(20)
 let testResourcesPath = "ably-common/test-resources/"
 let echoServerAddress = "https://echo.ably.io/createJWT"
 
+func uniqueChannelName(prefix: String = "",
+                       testIdentifier: String = #function,
+                       timestamp: TimeInterval = Date.timeIntervalSinceReferenceDate) -> String {
+    let platform: String
+#if targetEnvironment(macCatalyst)
+    platform = "macCatalyst"
+#elseif os(OSX)
+    platform = "OSX"
+#elseif os(iOS)
+    platform = "iOS"
+#elseif os(tvOS)
+    platform = "tvOS"
+#elseif os(watchOS)
+    platform = "watchOS"
+#else
+    platform = "Unknown"
+#endif
+    return "\(prefix)-\(platform)-\(testIdentifier.replacingOccurrences(of: "()", with: ""))-\(timestamp)"
```

Comment:
Do you have an idea of the resolution of this timestamp? We have some tests that generate two channel names in a row. In that case, I think that even millisecond resolution might not guarantee uniqueness. Could we put a UUID in here somewhere?

Reply:
>two channel names in a row
I tried to avoid situations like that, couldn't you point them out, please..

Reply:
> couldn't you point them out, please...

I'm also curious :) because this shouldn't happen anywhere

Reply:
I think I was wrong – I had a look through the diff again and I'm struggling to find any!

Reply:
What we _do_ have, however, are a few places where we generate multiple channel names in a row, and where we rely on the `prefix` parameter to ensure that these names are unique, e.g. [here](https://github.com/ably/ably-cocoa/blob/7ee8a3dbcfc984a10788f1ace810c66b330af690/Spec/Tests/RealtimeClientChannelTests.swift#L643-L649) and [here](https://github.com/ably/ably-cocoa/blob/7ee8a3dbcfc984a10788f1ace810c66b330af690/Spec/Tests/RealtimeClientChannelTests.swift#L832-L838) and [here](https://github.com/ably/ably-cocoa/blob/7ee8a3dbcfc984a10788f1ace810c66b330af690/Spec/Tests/RealtimeClientChannelTests.swift#L2120-L2121).

I think that since the method is called `uniqueChannelName`, it really should always return a unique value, not "a unique value as long as you either call the method in different milliseconds or pass a different prefix", else it's going to be confusing for people reading the tests, and using this method in the future. So, yeah, I'd say to put a UUID in there.

Reply:
6bfb4bab84ff44ffac3d2a36eb67c3e5cf877eef

Reply:
I'll let @lawrence-forooghian close this conversation, [as etiquette states](https://github.com/ably/engineering/blob/main/sdk/pull-requests.md#conversations), but I just wanted to add that I fully approve of adding UUID.